dashboard/src/components/DeploymentForm/BasicDeploymentForm/BasicDeploymentForm.tsx

```diff
@@ -56,6 +65,23 @@ class BasicDeploymentForm extends React.Component<IBasicDeploymentFormProps> {
             param={param}
           />
         );
+      case EXTERNAL_DB_PARAM_NAME:
+        return (
+          <ExternalDatabaseSection
+            label="External Database Details"
+            handleBasicFormParamChange={this.props.handleBasicFormParamChange}
+            key={id}
+            externalDatabaseParams={this.filterDatabaseParams()}
+          />
+        );
+      case EXTERNAL_DB_HOST_PARAM_NAME:
+      case EXTERNAL_DB_USER_PARAM_NAME:
+      case EXTERNAL_DB_PASSWORD_PARAM_NAME:
+      case EXTERNAL_DB_NAME_PARAM_NAME:
+      case EXTERNAL_DB_PORT_PARAM_NAME:
+      case USE_SELF_HOSTED_DB_PARAM_NAME:
+        // Handled within ExternalDabataseSection
+        break;
```

Comment:
> So the schema might identify one, some or all of the individual external db fields - and you're using this param name as a "parent" or something?

That's correct.

> Also, the schema may or may not define a use self-hosted db toggle right?

In theory it should. There should be an option of enabling/disabling the external database. If not, those would be normal parameters.

> So I think what you're aiming for (and achieving) with the above code, is something like: "if the chart schema defines the external db param name, display the ExternalDatabaseSection with the provided params (and remove the placeholder)." ?

Right (but I am not removing the placeholder?)

> rather than what you're dealing with here which is either a 1:1 or M:1 mapping? Let me know what you think. I worry that this extra work here will be repeated if/when other grouped params are added.

Good point. Yes, probably we should render that dependency from the schema parsing instead than from here. Let me give that a try.

> (the other thing that's a little confusing IMO is that you've named the component ExternalDatabaseSection but it includes the toggle to explicitly not use an external db... I'd rename it DatabaseSection or DatabaseDetailsSection?)

Got it.

Reply:
So this is possible, the only problem is that the parameter that select if the external database is being used or not is handled in a different object `mariadb.enabled`. This causes that that element is handled a bit different.

In any case this has been improved now. Let me know what you think.

Reply:
Looking much better, thanks! Note that we still have one switch case above with the same smell - I've put some thoughts below.